### ISSUE TITLE
Publishing versions 0.4 of HAL and 0.25.0 of PAC

### ISSUE THREAD

**Issue by jelacious** [atsams-rs/atsamx7x-rust#34]
Many of the crates in this megaproject has seen a lot of changes since last publication on crates.io. I just assume this has slipped the mind of the maintainers. Latest version (0.1.0) was published in March. It'd be nice to have the 0.3 versions of the HAL and board crates be published, if they are ready ofc. All the peripheral access APIs have a version of 0.25.0 in their Cargo.toml files but the latest version published is 0.21.0 also back in March.

Looking forward to getting acquainted with this. Keep it up <3

**Comment by michalfita** [comment]
It's work in progress, but I have limited capacity to touch this ATM. It should happen before end of this year.

PAC versioning is weird at the moment. We base it on minor version of `svd2rust`, so fluctuates as we're getting new releases.

**Comment by jelacious** [comment]
Ah, that makes sense. 

No rush, will just pin myself against the repo itself for now.

**Comment by tmplt** [comment]
> It'd be nice to have the 0.3 versions of the HAL and board crates be published

We don't offer proper BSPs yet, but I don't think it requires a lot of work to get there.

**Comment by michalfita** [comment]
@jelacious I let myself to use this issue to address the release.

- [x] Lying foundations
- [x] Tagging `pac/v0.25.0`
- [x] Publishing PAC crates
- [ ] Tagging `hal/v0.4.0`
- [ ] Publishing HAL crates

**Comment by michalfita** [comment]
PAC crates v0.25.0 released to [crates.io](https://crates.io/keywords/atsam).

**Comment by tmplt** [comment]
@michalfita, I suggest adding me as co-owner of repo and crates so there are multiple people that can handle releases.

**Comment by michalfita** [comment]
Not finished.
